### PR TITLE
Wkd advanced

### PR DESCRIPTION
**Pull request by c8k**
Added the advanced method of WKD and the unit test for the url. Issue #774

### CODE REVIEW

**Comment by c8k** [comment]
@toberndo just wanted to say that this pull request is ready to be merged.

File: src/firefox/manifest.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
     "nativeMessaging",
     "storage",
     "tabs",
-    "webNavigation"
+    "webNavigation",
+    "dns"
```
Comment (toberndo) [inline src/firefox/manifest.json:40]: This list is alphabetically ordered

File: src/modules/wkdLocate.js [hunk 2]
```diff
@@ -61,24 +61,61 @@ export async function lookup(email) {
...
-  // Impose a size limit and timeout similar to that of gnupg.
-  const data = await timeout(TIMEOUT * 1000, window.fetch(url)).then(
-    res => sizeLimitResponse(res, SIZE_LIMIT * 1024));
+  /** For the WKD standard (draft version 13, https://datatracker.ietf.org/doc/draft-koch-openpgp-webkey-service/13/) it is important to check, if the subdomain openpgpkey exists. Only in that
+  * case we should use the advanced method. The optimal way to do this check, is to try to
+  * resolve the DNS name. On the 21st of November 2021 only Firefox (since version 60) supported the method dns.resolve()
+  * (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/dns/resolve).
+  * That's why we check first, if this method exists.
+  */
+  if (typeof browser !== 'undefined' && typeof(browser.dns.resolve) === 'function') {
```
Comment (toberndo) [inline src/modules/wkdLocate.js:72]: Check on existence of browser not required as we use webextension-polyfill

File: src/firefox/manifest.json (hunk 1 above)
Comment (toberndo) [inline src/firefox/manifest.json:40]: We need to make sure that the user is not asked for additional permissions on extension update. Does the dns permission trigger such a request for additional permissions?

Comment (c8k) [reply src/firefox/manifest.json]: Is fixed now

Comment (c8k) [reply src/firefox/manifest.json]: The permissions Firefox is asking for are the same as before making the change.

File: src/modules/wkdLocate.js (hunk 2 above)
Comment (c8k) [reply src/modules/wkdLocate.js]: I've changed that part

**Comment by c8k** [comment]
@toberndo thanks for your review! Is there anything else I can do to make this pull request ready for a merge?

**Pull request merged**